plugin/cancel/cancel.py

```diff
@@ -1,20 +1,70 @@
 # to disable command cancellation, comment out this entire file.
 # you may also wish to adjust the commands in misc/cancel.talon.
-from talon import actions, speech_system
+
+import time
+
+from talon import Context, Module, actions, speech_system
+from talon.grammar import Phrase
 
 # To change the phrase used to cancel commands, you must also adjust misc/cancel.talon
 cancel_phrase = "cancel cancel".split()
 
+mod = Module()
+ctx = Context()
+
+ts_threshold: float = 0
+
+
+@ctx.action_class("speech")
+class SpeechActions:
+    # When Talon wakes we set the timestamp threshold. On the next command we
+    # will compare the phrase timestamp to the threshold and cancel any phrase
+    # started before wakeup. This is to prevent speech said before wake-up to
+    # be interpreted as a command if the user wakes Talon using a noise or
+    # keypress.
+    def enable():
+        actions.user.cancel_current_phrase()
+        actions.next()
+
+
+@mod.action_class
+class Actions:
+    def cancel_current_phrase():
+        """Cancel/abort current spoken phrase"""
```

Comment:
Super minor nitpick: do we see a use case for overriding or calling this action elsewhere?  I wonder if ts_threshold should just be set directly in enable()

'Cause I don't think folks would be able to wire up a command to cancel things using this action, as the the timestamp would be set after the command is executed.

Reply:
Agreed, for example if you use `speech.toggle` it'll still call `speech.enable`, so there's no need to separate it out.

Reply:
I actually use that in my own code when I toggle mute in discord. I thought that maybe someone else might have a similar need and having that extra action doesn't really hurt. 
https://github.com/AndreasArvidsson/andreas-talon/blob/53ba2db930f648683092df05f1af80de3e2ac367/games/games.py#L44

Reply:
OK, I don't feel too strongly about this, sounds like a reasonable argument to keep it separate.

Reply:
It's probably an uncommon use case so I could go either way

Reply:
It seems reasonable to keep it then. Thanks